rust_src/src/xsettings.rs

```diff
@@ -32,4 +33,20 @@ pub fn font_get_system_font() -> LispObject {
     }
 }
 
+/// Get the system tool bar style.
+/// If no system tool bar style is known, return `tool-bar-style' if set to a
+/// known style.  Otherwise return image.
+#[lisp_fn]
+pub fn tool_bar_get_system_style() -> LispObject {
+    match unsafe { globals.Vtool_bar_style } {
```

Comment:
Does work?

```
match unsafe { globals.Vtool_bar_style } {
  Qnil => unsafe { current_tool_bar_style },
  style @ Qimage | Qtext | Qboth | Qboth_horiz | Qtext_image_horiz => style,
  _ => Qimage,
}
```

Reply:
After a bit of experimentation, this is what I have:
```
    match unsafe { globals.Vtool_bar_style } {
        Qnil => unsafe { current_tool_bar_style },
        style @ Qimage
        | style @ Qtext
        | style @ Qboth
        | style @ Qboth_horiz
        | style @ Qtext_image_horiz => style,
        _ => Qimage,
    }
```
Which do you prefer?

Reply:
Yeah. Yuck:
> If you use @ with |, you need to make sure the name is bound in each part of the pattern:

I am inclined to leave the @ binding off and simply return `unsafe { globals.Vtool_bar_style }`.

Reply:
I took the bindings off, and added a local `style` variable to the top to be used both in the match and in that return branch.